analyzers/tests/SonarAnalyzer.Test/Common/DisjointSetsTest.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace SonarAnalyzer.Test.Common;
+
+[TestClass]
+public class DisjointSetsTest
+{
+    [TestMethod]
+    public void FindRootAndUnion_AreConsistent()
+    {
+        var elements = Enumerable.Range(1, 5).Select(x => x.ToString());
```

Comment:
Can you give an example how would you do that?
- One test use the elements directly, so you can't return the `DisjointSets`.
- If you return the elements, the method becomes `Enumerable.Range(1, max).Select(x => x.ToString())`
- Max needs to be passed anyway, because the last of the three methods takes a different range
- The method can't be reused for the new `GetAllSets_ReturnsSortedSets` test, because this test requires the input enumerable not to be in ascending order

Reply:
I was wondering if it was possible during test setup to create one enumerable for all the tests. And use the same one everywhere.
If not please ignore this comment.

Reply:
I have changed the assertions of the `GetAllSetsAndUnion_AreConsistent,` which was based on a list of 5 elements, to make it work with a list of 6 integers, which is required to test the merge of the 1-height tree and 2-height tree in `GetAllSetsAndUnion_OfNestedTrees.`
That's the only way I can see I can make this work. I can't do the opposite because I don't have enough elements to make two independent trees of height 1 and 2, where the tree of height 2 comes from a merge of trees of height 1.
As mentioned above, `GetAllSets_ReturnsSortedSets` requires a different order of elements, and that's why its input is different.